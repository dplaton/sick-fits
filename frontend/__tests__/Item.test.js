import ItemComponent from '../components/Item';
import {shallow} from 'enzyme'
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: 'ABC123',
    title: 'A fake item',
    price: 1000,
    description: 'This is a fake item',
    image:'shoe.jpg',
    largeImage:'a_larger_shoe.jpg'
}

describe('<Item/>', () => {

    it('renders a component that matches the snapshot', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    /* 
    it('renders the image properly', () => {
        const wrapper = shallow(<Item item={fakeItem}/>);

        const img = wrapper.find('img');
        expect(img.props().src).toBe(fakeItem.image);
        expect(img.props().alt).toBe(fakeItem.title);
    })

    it('renders the metadata properly', () => {
        const wrapper = shallow(<Item item={fakeItem}/>);

        const PriceTag = wrapper.find('PriceTag');
        expect(PriceTag.children().text()).toBe('$10');
        expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
        expect(wrapper.find('p').text()).toBe(fakeItem.description);        
    });

    it('renders the buttons properly', ()=>{
        const wrapper = shallow(<Item item={fakeItem}/>);
        
        const ButtonList = wrapper.find('.buttonList');
        expect(ButtonList.children()).toHaveLength(3);
        expect(ButtonList.find('Link').exists()).toBe(true);
        expect(ButtonList.find('AddToCart').exists()).toBe(true);
        expect(ButtonList.find('DeleteItem').exists()).toBe(true);
    })
    */
})