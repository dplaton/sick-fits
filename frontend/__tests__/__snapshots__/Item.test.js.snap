// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders a component that matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A fake item"
    src="shoe.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A fake item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $10
  </PriceTag>
  <p>
    This is a fake item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
